prow/flagutil/kubernetes_cluster_clients.go

```diff
@@ -409,6 +429,79 @@ func (o *KubernetesOptions) BuildClusterManagers(dryRun bool, callBack func(), o
 	return res, aggregatedErr
 }
 
+// checkAuthorizations checks if we are able to list pods using the provided
+// build cluster configuration.
+func checkAuthorizations(config rest.Config, namespace string) error {
+	client, err := authorizationv1.NewForConfig(&config)
+	if err != nil {
+		return err
+	}
+	sari := client.SubjectAccessReviews()
+
+	verbs := []string{
+		"create",
+		"delete",
+		"list",
+		"watch",
+		"get",
+		"patch",
+	}
+
+	// Unfortunately we have to do multiple API requests because there is no way
+	// to check for multiple verbs on a resource at once. The closest
+	// alternative is the "*" wildcard verb, but that appears to be overbroad
+	// and fails on the integration test cluster. The approach we take here is
+	// essentially equivalent to the following kubectl command:
+	//
+	// 	 $ cat <<EOF | kubectl --context=kind-kind-prow-integration create -f - -v 8
+	//   apiVersion: authorization.k8s.io/v1
+	//   kind: SubjectAccessReview
+	//   spec:
+	//     resourceAttributes:
+	//       resource: pods
+	//       verb: list # also test for get, create, etc
+	//       namespace: test-pods
+	//     user: system:serviceaccount:default:prow-controller-manager
+	//   EOF
+	//
+	//  The difference in our case is that (1) we are running the below check
+	//  *inside* the main service cluster itself, and (2) we are running the
+	//  check against an entirely different build cluster.
+	//
+	// See
+	// https://kubernetes.io/docs/reference/access-authn-authz/authorization/#checking-api-access
+	// for more information.
+	for _, verb := range verbs {
+		sar := k8sauthorizationv1.SubjectAccessReview{
+			Spec: k8sauthorizationv1.SubjectAccessReviewSpec{
+				ResourceAttributes: &k8sauthorizationv1.ResourceAttributes{
+					Namespace: namespace,
+					Verb:      verb,
+					Resource:  "pods",
+				},
+				// FIXME: In production (for GKE at least), this should look
+				// more like "foo@project.iam.gserviceaccount.com" because the
+				// build clusters will grant authorization to PCM's Google
+				// Service Account. This is because from the build cluster's
+				// perspective, Prow (PCM) will connect to it as a GSA via
+				// Workload Identity. So we need a way to pass in this "foo@..."
+				// GSA here from a config somewhere to here.
+				User: "system:serviceaccount:default:prow-controller-manager",
```

Comment:
@cjwagner Does this sound right to you? I'd like a second pair of eyes for this assumption.

Reply:
> Maybe the right way forward is to just look up the `annotations` field for `iam.gke.io/gcp-service-account` ?

That would work for us, but that is GKE specific.  If we were ok with adding GKE specifics we could just ask `gcloud` what our identity is, but I think it is probably more appropriate to just add a config field to the `plank` config like you suggested in the comment.
That will be provider agnostic, simpler to implement, and would also make this new validation opt-in which is safer at first.

Reply:
Cool, good to know I was on the right track. Will update this PR to add this in.

Reply:
I found a simpler way --- just use SelfSubjectAccessReview instead (this way we essentially defer whatever the `User` should be to the kubeconfig we're using for the build cluster).